src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -230,5 +230,30 @@ private enum WM : uint
             CLOSE = 0x0010
         }
 #endif
+
+        /// <summary>Gets <see cref="Thread.CurrentPrincipal"/> or <see langword="null" /> if the current platform does not support it.</summary>
+        public static System.Security.Principal.IPrincipal GetCurrentThreadPrincipal()
+        {
+            try
+            {
+                return Thread.CurrentPrincipal;
+            }
+            catch (PlatformNotSupportedException) //E.g. Mono.WASM
+            {
+                return null;
+            }
+        }
+
+        /// <summary>Sets <see cref="Thread.CurrentPrincipal"/> if current platform supports it.</summary>
+        /// <param name="principal">Value to set. If current platform does not support <see cref="Thread.CurrentPrincipal"/> only allwoed value is null.</param>
```

Comment:
A small typo and a bit of rephrasing.
```suggestion
        /// <param name="principal">Value to set. If the current platform does not support <see cref="Thread.CurrentPrincipal"/> then the only allowed value is null.</param>
```
Perhaps also replace `null` with `<see langword="null" />` ? (and similar for the summary of `SandboxedThreadState.Principal`).